packages/astro/src/config/index.ts

```diff
@@ -7,6 +7,7 @@ import type {
 	SessionDriverName,
 } from '../types/public/config.js';
 import { createDevelopmentManifest } from '../vite-plugin-astro-server/plugin.js';
+import type { BuiltInProvider, FontFamily, FontProvider } from '../assets/fonts/types.js';
 
```

Comment:
These types **must** be exported from `types/public/config.js` 

As part of Astro v5, we did a big refactor and moved all types inside `types/public` and `type/private`, so we know whether some types are public (for users to use) or not

Reply:
I think these types should remain private, because we have a `defineFontProvider` type utility. What do you think?

Reply:
> What do you think?

It's very difficult to predict how users will use our APIs, but users can definitely do something like:

```js

function highOrderDefineProvider() {
	return something;
}

export function defineProvider() {
	return highOrderDefineProvider()
}
```

Plus, if a user uses `--isolatedModules` from typescript, they are forced by the compiler to have types everywhere, even then they can be inferred by the compiler, which means that that uses are forced to have type, which we should provide.